apps/judicial-system/web/src/routes/Court/InvestigationCase/HearingArrangements/HearingArrangements.tsx

```diff
@@ -96,38 +97,45 @@ const HearingArrangements = () => {
     workingCase,
   ])
 
-  const handleNextButtonClick = useCallback(() => {
-    const hasSentNotification = workingCase?.notifications?.find(
-      (notification) => notification.type === NotificationType.COURT_DATE,
-    )
+  const handleNavigationTo = useCallback(
+    async (destination: string) => {
+      const hasSentNotification = workingCase?.notifications?.find(
+        (notification) => notification.type === NotificationType.COURT_DATE,
+      )
+
+      setAndSendCaseToServer(
+        [
+          {
+            courtDate: courtDate
+              ? formatDateForServer(new Date(courtDate))
+              : undefined,
+            force: true,
+          },
+        ],
+        workingCase,
+        setWorkingCase,
+      )
 
-    setAndSendCaseToServer(
-      [
-        {
-          courtDate: courtDate
-            ? formatDateForServer(new Date(courtDate))
-            : undefined,
-          force: true,
-        },
-      ],
+      if (hasSentNotification && !courtDateHasChanged) {
+        router.push(`${destination}/${workingCase.id}`)
+      } else {
+        setNextRoute(`${destination}/${workingCase.id}`)
+        setModalVisible(true)
```

Comment:
Consider refactoring the use of these `nextRoute` states to be a part of setting the modal to visible. That way you will never render a modal that expects a destination route without one. This applies to all use-cases of this `nextRoute` states.